### ISSUE TITLE
Keyboard shortcut conflicting with Chrome "View Source" under macOS

### ISSUE THREAD

**Issue by schroedermatthias** [canonical/multipass#1311]
> "Use a keyboard shortcut (Ctrl+Alt+U or ⌘⌥U) to open a shell in your primary instance" 

introduced in 1.0.0. This is basically a very nice feature, but it should not be hardcoded but configurable.

Web developers tend to use "View Source" in Chrome, which has the exact same shortcut and is now conflicting. Multipass has now precedence and the shortcut in Chrome is no longer working.

**Comment by matthewpaulthomas** [comment]
Bother. I was going by [Microsoft’s list of Windows keyboard shortcuts](https://support.microsoft.com/en-gb/help/12445/windows-keyboard-shortcuts). and I didn’t consider that a third-party app might use Ctrl+Alt+anything.

Any suggestions for a replacement?

**Comment by ricab** [comment]
I'm not sure we could have chosen a better default. I don't have a suggestion for that, but I think this should be configurable. 

Whatever shortcut we choose, there can always be something else using the same. For development, in particular, the keyboard tends to be used up a lot. For instance, CLion alone probably uses most ctrl+alt+\<key\> combinations (Ctrl+Alt+U included). And people have their own custom shortcuts. There are only so many combinations and collision is unavoidable.

**Comment by Saviq** [comment]
+1 on this being configurable, fortunately it's simply a string, but we may need to translate some combinations of "Cmd", "Option", "⌘⌥" into what Qt expects.

**Comment by matthewpaulthomas** [comment]
In [multipass-design#46](https://github.com/canonical-web-and-design/multipass-design/pull/46) I’ve added this to the settings design:

![settings-general](https://user-images.githubusercontent.com/19801137/75227780-bc167f00-57a6-11ea-8e97-c679ccbd82b6.png)

**Comment by gwatts** [comment]
If this is going to take a while to address via a preference, could it either be removed entirely in the interim or provide some manual way to disable?

It's definitely tough to pick a global shortcut that doesn't collide with something, unfortunately this one collides with view-source in Chrome, Mail.app, etc. which I use all the time :-(

**Comment by RyanRoberts** [comment]
Just adding to the comments above, this being hard coded is incredibly annoying when you're used to using it to view source in Chrome and Firefox.

**Comment by dennisfrank** [comment]
Oh please yes, remove this shortcut! I installed Multipass today for the first time and every few minutes it disrupts my workflow when working with different browsers. :( 

(It’s not a good practice to register global, OS-wide shortcuts without an option to change it. I also use other apps where app specific `⌘⌥U` shortcuts are no longer usable when Multipass is running.)

**Comment by dennisfrank** [comment]
As a temporary workaround I used to [BetterTouchTool](https://folivora.ai/) to overwrite <kbd>⌘</kbd>+<kbd>⌥</kbd>+<kbd>U</kbd> globally and added some menubar item triggers for some browsers:

![Screen_Shot_2020-05-22_at_19 20 14](https://user-images.githubusercontent.com/65968/82693591-18802200-9c62-11ea-9c0f-2e9974b0281c.png)

Of course this does not solve the fundamental problem. But maybe it helps others as a short-term workaround.

**Comment by Saviq** [comment]
Hey all, sorry about stomping on your shortcuts, we'll have a fix for this asap. Note that Multipass will still function without the system menu item, there's a "Autostart on login" item under the About menu that you can disable. You can then run Multipass manually from Launchpad to re-enable it.

**Comment by gwatts** [comment]
Thanks for addressing this, @Saviq & @ricab; the update works great and i've safely remapped the hotkey to something less disruptive 🎉  😄 

For anyone else stumbling across this ticket, see https://discourse.ubuntu.com/t/multipass-set-command/13734

TLDR: `multipass set client.gui.hotkey="opt+shift+m"` or `multipass set client.gui.hotkey=""` if you don't want the hotkey at all (supported by multipass 1.4.0 or later)

**Comment by AnsisMalins** [comment]
As a general rule when giving global Alt+Shift shortcuts to your applications, please limit yourself to letters that don't have diacritics in any language. For example, Latvian keyboard layout does ēūīōāšģķļžčņ using right Alt plus the letter. Right Alt is the same as Alt+Shift. So please don't put any Alt+Shift shortcuts on those letters.

**Comment by Saviq** [comment]
@AnsisMalins that would likely exclude all the letters available on the keyboard… The shortcut uses left Alt (_not_ AltGr, which is used for diacritics), so that shouldn't be a problem, unless something is wrong. In which case, please open a new issue.
